core/build.gradle.kts

```diff
@@ -41,6 +41,7 @@ dependencies {
 
     api("com.fasterxml.jackson.core:jackson-annotations")
     api("org.apache.calcite.avatica:avatica-core")
+    api("org.apiguardian:apiguardian-api")
```

Comment:
this seems to be another feature that is unrelated with this pr

Reply:
It is related. The dependency is needed to add `@API(since = "1.22",` annotation to the methods that are added in this PR.

Reply:
Can we write our own annotation ? I would expect to not introduce a dependency just for a annotation, using our own annotations keep better compatibility and reduce conflicts.

Reply:
Do you expect the same annotation to be used in Avatica and Calcite?
Does that mean the annotation should be declared in Avatica, and can be used in Calcite only after Avatica is released? It does sound too complicated.

On the other hand, https://github.com/apiguardian-team/apiguardian is a more-or-less clear dependency that has already been tested (e.g. it is used in JUnit5).

In other words, `apiguardian-api` is already available (and it can be easily used by other projects for consistent API declaration!) while Calcite's annotation is nowhere near.